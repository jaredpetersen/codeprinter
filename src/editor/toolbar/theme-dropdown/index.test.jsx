import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import ThemeDropdown from './index';
import { DropdownItem } from 'reactstrap';

describe('Editor Toolbar ThemeDropdown', () => {
  it('renders without crashing', () => {
    const themes = ['GitHub', 'VS', 'Xcode'];
    shallow(<ThemeDropdown themes={themes} active={themes[0]} onSelect={() => {}} />);
  });

  it('renders all of the themes as dropdown options', () => {
    const themes = ['GitHub', 'VS', 'Xcode'];
    const tree = renderer.create(<ThemeDropdown themes={themes} active={themes[1]} onSelect={() => {}} />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('calls the onSelect function on selecting a dropdown option ', () => {
    const themes = ['GitHub', 'VS', 'Xcode'];
    const onSelect = jest.fn();
    const themeDropdown = shallow(<ThemeDropdown themes={themes} active={themes[0]} onSelect={onSelect} />);

    themeDropdown
      .find(DropdownItem)
      .last()
      .simulate('click');

    expect(onSelect).toHaveBeenCalledTimes(1);
  });
});
