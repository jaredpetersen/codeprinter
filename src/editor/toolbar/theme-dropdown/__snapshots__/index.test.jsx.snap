// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Editor Toolbar ThemeDropdown renders all of the themes as dropdown options 1`] = `
<div
  className="dropdown"
  onKeyDown={[Function]}
>
  <button
    aria-expanded={false}
    aria-haspopup={true}
    aria-label={null}
    className="dropdown-toggle-select btn btn-outline-dropdown-toggle-select btn-block"
    onClick={[Function]}
    style={
      Object {
        "textAlign": "left",
      }
    }
    type="button"
  >
    VS
  </button>
  <div
    aria-hidden={true}
    className="dropdown-menu"
    role="menu"
    tabIndex="-1"
  >
    <button
      className="dropdown-item"
      onClick={[Function]}
      role="menuitem"
      tabIndex="0"
      type="button"
    >
      GitHub
    </button>
    <button
      className="dropdown-item active"
      onClick={[Function]}
      role="menuitem"
      tabIndex="0"
      type="button"
    >
      VS
    </button>
    <button
      className="dropdown-item"
      onClick={[Function]}
      role="menuitem"
      tabIndex="0"
      type="button"
    >
      Xcode
    </button>
  </div>
</div>
`;
